WordPress/gutenberg
# Align data view icon usage

## What?
Update icon usage in data view UIs.

## Why?
In data views, the icon used in menu items and the View options button are supposed to reflect the layout associated with each view. There are some inconsistencies in the current implementation that break that intended alignment:

| | List layout icon | Grid layout icon | Side by side layout icon | 
| --- | --- | --- | --- |
| View button | `blockTable` | `grid` | `columns` |
| Menu item | `page` | `columns` | `columns` |

This PR updates each element to use the following icons:

| | List layout icon | Grid layout icon | Side by side layout icon | 
| --- | --- | --- | --- |
| View button | `formatListBullets ` | `category` | `columns` |
| Menu item | `formatListBullets ` | `category` | `columns` |


### Before
<img width="1458" alt="Screenshot 2023-11-28 at 16 37 42" src="https://github.com/WordPress/gutenberg/assets/846565/0a489fa4-58b3-4468-a876-ba369b8db5b1">


### After
<img width="1460" alt="Screenshot 2023-11-28 at 16 43 21" src="https://github.com/WordPress/gutenberg/assets/846565/ed67e12f-d2e2-48ee-97df-9daa7b3c4afa">


## Testing Instructions
1. Enable the data view experiment
2. Navigate to Manage all pages in the site editor
3. Create a custom view that uses the grid layout
4. Ensure that the icon in the active menu item always matches the icon in the View options button

## Caveat
`formatListBullets` is by no means the perfect icon for the list layout. However, we probably need a new icon for this use case, one that potentially harmonises better with `category`. I intend to explore that in a dedicated follow up, this PR is purely about ensuring consistent behavior between these related UI elements.
